### PR TITLE
[WIP] github actions rust cache

### PR DESCRIPTION
**Pull request by jbolda**
**What kind of change does this PR introduce?** (check at least one)
- [ ] Bugfix
- [ ] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe: github actions

**Does this PR introduce a breaking change?** (check one)
- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**
- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Finishing up the cache uploads by setting the ext, then caching some of the Rust items.

### CODE REVIEW

File: .github/workflows/build-smoke-tests.yml [hunk 1]
```diff
@@ -59,8 +77,12 @@ jobs:
             folder: vue/quasar-app
             executable: app
         include:
+          - platform: ubuntu-latest
+            ext: _0.1.0_amd64.deb
+          - platform: macos-latest
+            ext: .app
```
Comment (nothingismagick) [inline .github/workflows/build-smoke-tests.yml]: we need to actually descend a bit deeper for macos - on my mac, they end up e.g. here:

`examples/vue/quasar-app/src-tauri/target/release/bundle/osx/app.app`

Comment (jbolda) [reply .github/workflows/build-smoke-tests.yml]: 👍 I added another property that we can specify the release folder. However, is there any way we can make this more standard between platforms? I expect cross-platform building will be a selling point and that consistency would be helpful.

Comment (nothingismagick) [reply .github/workflows/build-smoke-tests.yml]: This is something to address in the cargo bundler - but yes it should definitely be possible.

**Comment by jbolda** [comment]
Still working out a couple last things, some tests are green that aren't working (which is fine because we wouldn't want failing cache to break the build).

I need to get the artifacts working for ubuntu still, and the rust caching isn't quite working yet.

**Pull request merged**